pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1014,27 +1010,64 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 	return nil
 }
 
-func calculateRequestedVCPUs(vmi *v1.VirtualMachineInstance) uint32 {
-	cores := uint32(0)
-	if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Cores != 0 {
+func getCPUTopology(vmi *v1.VirtualMachineInstance) *CPUTopology {
+	return &CPUTopology{
+		Sockets: calculateRequestedSockets(vmi),
+		Cores:   calculateRequestedCPUs(vmi),
+		Threads: calculateRequestedThreads(vmi),
+	}
+}
+
+func calculateRequestedSockets(vmi *v1.VirtualMachineInstance) uint32 {
+	sockets := uint32(1)
+	cpu := vmi.Spec.Domain.CPU
+	if cpu.Sockets != 0 {
+		return cpu.Sockets
+	}
+
+	if cpu.Cores == 0 && cpu.Threads == 0 {
```

Comment:
What happens if one of them is 0?

How does this affect `calculateRequestedVCPUs`?

Reply:
if one of them is 0, then functions calculateRequestedThreads or calculateRequestedCPUs will change the value to 1 and the other one will remain the same.